torch/nn/parallel/distributed.py

```diff
@@ -464,6 +473,11 @@ def __init__(self, module, device_ids=None,
         if self.device_ids is not None and len(self.device_ids) > 1:
             dist._verify_replicas_within_process(parameters, expect_sparse_gradient)
         dist._verify_model_across_ranks(self.process_group, parameters)
+        if _GLOO_AVAILABLE and dist._get_debug_mode() != dist._DistributedDebugLevel.OFF:
+            # Run monitored barrier in debug mode. This will ensure all ranks
+            # have same model shapes before continuing with init/training.
+            dist.monitored_barrier(group=self._cpu_pg, timeout=30)
```

Comment:
Are you planning to call dist.monitored_barrier after every collective call in DDP debug mode in following up diffs as well?

Reply:
Yes, but the current plan for enabling that is only in DETAIL debug mode, whereas this one will run in INFO or DETAIL. I think it's worth having monitored barrier here for INFO as we only do it in DDP init, doing it for every collective comm. call in INFO mode as well is quite expensive IMO and should be saved for the most verbose debug option.